activerecord/lib/active_record/associations.rb

```diff
@@ -229,7 +229,16 @@ def association(name) #:nodoc:
       association = association_instance_get(name)
 
       if association.nil?
-        unless reflection = self.class._reflect_on_association(name)
+        reflection = self.class._reflect_on_association(name)
```

Comment:
Yes that is probably the level to push this to, but we should have tests
for all the visible effects of this regardless of where it's implemented

On Wed, May 11, 2016, 10:33 PM Jon Moss <notifications@github.com> wrote:

> In activerecord/lib/active_record/associations.rb
> <https://github.com/rails/rails/pull/24719#discussion_r62957427>:
>
> > @@ -229,7 +229,16 @@ def association(name) #:nodoc:
> >        association = association_instance_get(name)
> >
> >        if association.nil?
> > -        unless reflection = self.class._reflect_on_association(name)
> > +        reflection = self.class._reflect_on_association(name)
>
> Should I push this down the stack to _reflect_on_association? The problem
> there, is that since this method is included via an AS::Concern, it
> doesn't know about the current class its dealing with, and thus wouldn't be
> able to construct the superclass tree, etc.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/24719/files/e22146a58b93d4a5ea53ef481387738f58bb1fd5#r62957427>
>